### PR TITLE
adding tests in sopt and core

### PR DESCRIPTION
**Pull request by nh13**

### CODE REVIEW

File: sopt/src/main/scala/dagr/sopt/ArgTokenCollator.scala [hunk 1]
```diff
@@ -71,7 +71,6 @@ class ArgTokenCollator(argTokenizer: ArgTokenizer) extends Iterator[Try[ArgOptio
       case Success(ArgOptionAndValue(name, value)) => values += value; Success(name)
       case Success(ArgValue(value)) => Failure(new OptionNameException(s"Illegal option: '$value'"))
       case Failure(ex) => Failure(ex)
-      case Success(tok) => throw new IllegalStateException(s"Unknown token: '${tok.getClass.getSimpleName}")
```
Comment (nh13) [inline sopt/src/main/scala/dagr/sopt/ArgTokenCollator.scala:74]: can't happen since `ArtTokenizer.Token`  is a sealed class.

**Comment by nh13** [comment]
@tfenne I am merging this as this is really not that exciting of a review, given its 90% tests.

File: core/src/main/scala/dagr/core/execsystem/ResourceSet.scala [hunk 2]
```diff
@@ -24,7 +24,7 @@
 package dagr.core.execsystem
 
 object ResourceSet {
-  def apply(that: ResourceSet): ResourceSet = new ResourceSet(cores = that.cores, memory = that.memory)
+  def apply(that: ResourceSet): ResourceSet = new ResourceSet(cores = Cores(that.cores.cores), memory = Memory(that.memory.value))
```
Comment (tfenne) [inline core/src/main/scala/dagr/core/execsystem/ResourceSet.scala:27]: Given that ResourceSet is immutable, perhaps it just makes sense to eliminate this apply method entirely?

File: core/src/main/scala/dagr/core/cmdline/parsing/ParsingUtil.scala [hunk 3]
```diff
@@ -35,15 +36,17 @@ import scala.collection.immutable.Map
...
 
 /** Variables and Methods to support command line parsing */
-private[parsing] object ParsingUtil {
+private[parsing] object ParsingUtil extends Configuration  {
   /** Gets the [[CLPAnnotation]] annotation on this class */
   def findClpAnnotation(clazz: Class[_]): Option[CLPAnnotation] = {
     ReflectionUtil.findJavaAnnotation(clazz, classOf[CLPAnnotation])
   }
 
+  private lazy val printColor: Boolean = optionallyConfigure[Boolean](Configuration.Keys.ColorStatus).getOrElse(false)
```
Comment (tfenne) [inline core/src/main/scala/dagr/core/cmdline/parsing/ParsingUtil.scala:45]: Two things here:
1) Perhaps we should just add this to the reference.conf and not use a default here?
2) If you want to keep the default in code there's also `configuration[Boolean](Configuration.Keys.ColorStatus, false)`

**Comment by tfenne** [comment]
Thanks for doing all this.  I read through anyway just to see what you were adding.  Made a couple of small suggestions you can chose to ignore if you like.

**Pull request merged**